homeassistant/components/fritzbox/__init__.py

```diff
@@ -84,8 +107,39 @@ async def async_setup_entry(hass, entry):
     except LoginError as err:
         raise ConfigEntryAuthFailed from err
 
-    hass.data.setdefault(DOMAIN, {CONF_CONNECTIONS: {}, CONF_DEVICES: set()})
-    hass.data[DOMAIN][CONF_CONNECTIONS][entry.entry_id] = fritz
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {
+        CONF_CONNECTIONS: fritz,
+        CONF_RETRY: False,
+    }
+
+    async def async_update_coordinator():
+        """Fetch all device data."""
+        try:
+            devices = await hass.async_add_executor_job(fritz.get_devices)
+        except requests.exceptions.HTTPError as ex:
+            if hass.data[DOMAIN][entry.entry_id][CONF_RETRY]:
+                raise ConfigEntryAuthFailed from ex
+            hass.data[DOMAIN][entry.entry_id][CONF_RETRY] = True
+            raise UpdateFailed(f"Fritzhome connection error: {ex}") from ex
```

Comment:
```suggestion
        try:
            devices = await hass.async_add_executor_job(fritz.get_devices)
        except requests.exceptions.HTTPError as ex:
            # If the device rebooted, login again
            try:
                await hass.async_add_executor_job(fritz.login)
            except requests.exceptions.HTTPError as ex:
                raise ConfigEntryAuthFailed from ex
            devices = await hass.async_add_executor_job(fritz.get_devices)
```

Reply:
oh yeah ... this one is quiet smarter than my solution ... thanks :+1: